controlpanel/frontend/views/dashboard.py

```diff
@@ -0,0 +1,303 @@
+# Third-party
+import sentry_sdk
+import structlog
+from django.contrib import messages
+from django.core.paginator import Paginator
+from django.db import transaction
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.template.defaultfilters import pluralize
+from django.urls import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.views.generic.base import RedirectView
+from django.views.generic.detail import DetailView, SingleObjectMixin
+from django.views.generic.edit import CreateView, DeleteView, FormMixin, UpdateView
+from django.views.generic.list import ListView
+from rules.contrib.views import PermissionRequiredMixin
+
+# First-party/Local
+from controlpanel.api.exceptions import DeleteCustomerError
+from controlpanel.api.models import Dashboard, DashboardDomain, DashboardViewer, User
+from controlpanel.frontend.forms import (
+    AddCustomersForm,
+    GrantDomainAccessForm,
+    RegisterDashboardForm,
+    RemoveCustomerByEmailForm,
+)
+from controlpanel.oidc import OIDCLoginRequiredMixin
+from controlpanel.utils import feature_flag_required
+
+log = structlog.getLogger(__name__)
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class DashboardList(OIDCLoginRequiredMixin, PermissionRequiredMixin, ListView):
+    context_object_name = "dashboards"
+    model = Dashboard
+    permission_required = "api.list_dashboard"
+    template_name = "dashboard-list.html"
+
+    def get_queryset(self):
+        return self.request.user.dashboards.all()
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class AdminDashboardList(DashboardList):
+    permission_required = "api.is_superuser"
+    template_name = "dashboard-admin-list.html"
+
+    def get_queryset(self):
+        return Dashboard.objects.all().prefetch_related("admins")
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class RegisterDashboard(OIDCLoginRequiredMixin, PermissionRequiredMixin, CreateView):
+    form_class = RegisterDashboardForm
+    model = Dashboard
+    permission_required = "api.register_dashboard"
+    template_name = "dashboard-register.html"
+
+    def get_success_url(self):
+        messages.success(
+            self.request,
+            f"Successfully registered {self.object.name} dashboard",
+        )
+        return reverse_lazy("manage-dashboard", kwargs={"pk": self.object.pk})
+
+    def form_valid(self, form):
+        # add dashboard, set creator as an admin and viewer
+        with transaction.atomic():
+            user = self.request.user
+            email = user.justice_email.lower()
+            dashboard = form.save()
+            dashboard.created_by = user
```

Comment:
You've already saved when`created_by` is assigned, so it wont be stored in the DB. To avoid saving twice, you could change to:
```
dashboard = form.save(commit=False)
dashboard.created_by = user
dashboard.save()
```